### PR TITLE
Add solidarity pledge

### LINKED ISSUE #2646: Add checkbox to mark solidarity pledges
Thanks for creating an issue! Please fill out this form so we can be sure to have all the information we need, and to minimize back and forth.

* **What are we trying to do?**

Solidarity pledges are an abortion funding term for 'pitching in some cash on a case another fund is handling'. The points of contact for these cases (iirc) tend to be other funds and not the patients themselves, and they tend to be fairly complex cases involving travel or expensive procedures.

* **What feature or behavior is this required for?**

Funds tend to keep track of these a little differently, and it would be helpful to have a simple indicator of whether or not a patient is a solidarity case.

* **How could we solve this issue? (Not knowing is okay!)**

Let's do the following:
* Add a `solidarity` field to the patient model
* Add a checkbox to the Abortion Information tab, preferably at the bottom of the right side column here
* Add this marker to the CSV export

image

* **Anything else?**

When DARIA was getting going, we called these external pledges; kind of cool that our system predates the current term of art, in a way. Someday we may consider renaming external pledges but I think we can worry about that later, so out of scope for this.

### PR DESCRIPTION
**Pull request by elimbaum**
I rule and have completed some work on Case Manager that's ready for review!

Does what it says on the tin!

This pull request makes the following changes:
* Adds solidarity boolean to patient model
* Updates view
* Adds to CSV export

(If there are changes to the views, please include a screenshot so we know what to look for!)

It relates to the following issue #s: 
* Fixes #2646 
* Bumps #Y

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.

### CODE REVIEW

**Comment by elimbaum** [comment]
tada! i'm getting the hang of these migrations ;)

**Comment by elimbaum** [comment]
Screen Shot 2022-07-28 at 21 59 17

new view screenshot!

**Review by colinxfleming — APPROVED** [review]
Very nicely done! Works pretty much perfectly. Two very minor things and then we can ship:

* The changelog entry shows up as translation missing. If you add a field at activerecord.attributes.patient.solidarity it'll show up cleanly.
image

* We have a patient edit system test that exercises these specific fields (control-f 'changing abortion information' in `update_patient_info_test.rb`). Would you mind extending our current test here to check this box and confirm it saves? 

After that we can ship. Thanks for the very fast turnaround on this!

**Comment by elimbaum** [comment]
you got it! was looking for a test like this but missed it here!

some weird system tests (`update_patient_info` L85 and L332) are failing but they're unrelated to this commit, and appear to be General Weirdness due to date entry. hopefully ok on CI.
\

File: config/locales/en.yml [hunk 1]
```diff
@@ -68,6 +69,7 @@ en:
         referred_to_clinic: Referred to clinic
         resolved_without_fund: Resolved without fund
         shared_flag: Shared Flag
+        solidarity: Solidarty Pledge
```
Comment (colinxfleming) [inline config/locales/en.yml:72]: lol
```suggestion
        solidarity: Solidarity Pledge
```

**Comment by elimbaum** [comment]
oops thx

**Comment by elimbaum** [comment]
oops thx

**Pull request merged**